lighthouse-core/runner.js

```diff
@@ -219,8 +219,11 @@ class Runner {
     const timingEntriesKeyValues = [
       ...timingEntriesFromArtifacts,
       ...timingEntriesFromRunner,
-      // As entries can share a name, dedupe based on the startTime timestamp
-    ].map(entry => /** @type {[number, PerformanceEntry]} */ ([entry.startTime, entry]));
+    ].map(entry => /** @type {[string, PerformanceEntry]} */ ([
+      // As entries can share a name and start time, dedupe based on the name, startTime and duration
+      `${entry.startTime}-${entry.name}-${entry.duration}`,
```

Comment:
This was a genuine issue I ran into and probably affects master as well. I'm thinking about down-streaming this fix and coming back to this PR when we have a `__arrayHasThisSomewhere` helper for smoke tests.